titan/agent.py

```diff
@@ -339,6 +311,37 @@ def get_partner(self, agent: "Agent") -> "Agent":
         else:
             raise ValueError("Agent must be in this relationship")
 
+    def get_number_of_sex_acts(self, rand_gen) -> int:
+        """
+        Number of sexActs an agent has done.
+
+        args:
+            rand_gen: random number generator (e.g. self.run_random in model)
+
+        returns:
+            number of sex acts
+        """
+        freq_params = self.agent1.location.params.partnership.sex.frequency[
+            self.bond_type
+        ]
```

Comment:
should this take into account agent2 at all?

Reply:
Probably! Surprisingly hard to figure out how that would work with bins.

Reply:
Ok, I'm opening an issue on this one, since I think it's going to take a bit of thought/discussion.